docs/src/main/sphinx/object-storage/metastores.md

```diff
@@ -689,7 +689,9 @@ properties:
 * - `iceberg.snowflake-catalog.user`
   - Snowflake user (required).
 * - `iceberg.snowflake-catalog.password`
-  - Snowflake password (required).
+  - Snowflake password (this or private-key required).
+* - `iceberg.snowflake-catalog.private-key`
+  - Private key content in PEM format (this or password required).
```

Comment:
Does it means Exactly one of the password or private key must be specified?
If so, please refer the https://github.com/trinodb/trino/blob/4f50ae66de7af866c8ad354499a2503c93027be5/lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3SecurityMappingConfig.java#L143
add the verification here

Reply:
@chenjian2664 Upon further analysis and testing, I figured that iceberg-jdbc connector does not support arbitrary types as jdbc property values. The `privateKey` need to be an instance of `java.security.PrivateKey`. Since the iceberg-jdbc's `JDBCClientPool` only supports Map<String, String> , we cannot implement this change for icerberg-snowflake as of now. Hence, removed all the changes related to that.

Reply:
I don't get you, which stops us doing this? is the implementation or the is that we shouldn't have such check?

Reply:
> we cannot implement this change

This comment was regarding the task as a whole. We cannot implement `privateKey` as a JVM property for iceberg-snowflake. Hence, your comment regarding adding the check is no longer valid. Because all the changes to this connector are now removed.

Reply:
@chenjian2664 Please do take a look and let me know if there are concerns. Additionally, for snowflake cloud tests, we need to add `SNOWFLAKE_PRIVATE_KEY` github secret and relevant changes in the cloud snowflake account used for CI testing.